src/component/toolbox/feature/DataView.js

```diff
@@ -426,13 +432,17 @@ function tryMergeDataOption(newData, originalData) {
     return zrUtil.map(newData, function (newVal, idx) {
         var original = originalData && originalData[idx];
         if (zrUtil.isObject(original) && !zrUtil.isArray(original)) {
-            if (zrUtil.isObject(newVal) && !zrUtil.isArray(newVal)) {
-                newVal = newVal.value;
+            var _newVal;
+            if (zrUtil.isArray(newVal) || zrUtil.isObject(newVal)) {
```

Comment:
Thanks for your contribution.

Could you explain the change here?

And please do not use `_` in variable names.

Reply:
Thanks for your review and sorry for the wrong variable name defined before.

The reason for making these changes is to ensure more compatibility at first.

For the case: **Delete any data item in any line in GridView**

In original logic context, it only handles the change of `value` but does not handle the change of `name`. Therefore, after any data item line is deleted, final merged result will ignore `name`.

An example:
**original data items:**
```js
[
    { name:  'USA', value: 15276 },
    { name:  'UK', value: 4855 }
]
```

**the first in original data items** 
`{ name: 'USA', value: 15276 }`

**the first data item after the old first data item was deleted(the second became the first)**
`{  name:  "UK", value: 4855  }`

After `tryMergeDataOption`, data items became
`[ { name: 'USA', value: 4855 } ]`

The result above is not obviously expected.
Both `name` and `value` of each data item should be handled and merged well.

By now, I think there should be some changes here.
```js
function tryMergeDataOption(newData, originalData) {
    return zrUtil.map(newData, function (newVal, idx) {
        var original = originalData && originalData[idx];
        if (zrUtil.isObject(original) && !zrUtil.isArray(original)) {
            var newValIsObject = zrUtil.isObject(newVal) && !zrUtil.isArray(newVal);
            if (!newValIsObject) {
                newVal = {
                    value: newVal
                };
            }
            if (original.name != null && newVal.name == null) {
                // original data has name but new data has no name
                delete original.name;
            }
            // Original data has option
            return zrUtil.defaults(newVal, original);
        }
        else {
            return newVal;
        }
    });
}
```

Finally, I found another bug that it will occur after editing the series name in DataView.

![image](https://user-images.githubusercontent.com/26999792/81162277-7ef91500-8fbf-11ea-8784-f356ccd8e74b.png)